sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -182,7 +182,8 @@ def do_resolve(value: Any, perform_apply: bool, exn: Optional[Exception]):
                 protect=opts.protect,
                 provider=resolver.provider_ref,
                 dependencies=resolver.dependencies,
-                propertyDependencies=property_dependencies
+                propertyDependencies=property_dependencies,
+                deleteBeforeReplace=opts.delete_before_replace
```

Comment:
can you add a test for this?

Reply:
sure--what sort? just a unit test to ensure that it's passed appropriately?

Reply:
yep, something simple like that.

Reply:
(I had a bug with `opts.protect` where it didn't get passed, so I'm hoping to get good test coverage on fundamentals like passing `opts` to the engine 😆 )